great_expectations/core/configuration.py

```diff
@@ -16,3 +30,13 @@ def __init__(self, id_: Optional[str] = None, name: Optional[str] = None) -> Non
         if name is not None:
             self.name = name
         super().__init__()
+
+    @classmethod
+    def dict_round_trip(cls, schema: Loadable, target: dict) -> dict:
+        """
+        Round trip a dictionary with a schema so that validation and serialization logic is applied.
+        Example: Swapping `id_` with `id`.
+        """
+        _loaded = schema.load(target)
+        _config = cls(**_loaded)
+        return _config.to_json_dict()
```

Comment:
I think the round trip we want is actually `load` and then `dump`? In speaking with Don, I think we want to avoid `to_json_dict` since it bypasses schema validation and the benefits of Marshmallow.

Reply:
Also instead of using this `Loadable` protocol, might it make more sense to have this be `Schema`? I understand the value in interfaces but I believe this value will always be a Marshmallow `Schema` right?

Reply:
I think they are both valid.
In this case, the only thing that matters is that the object has a `.load()` method that returns a dict-like or `Mapping` object, it doesn't actually need to be a Marshmallow `Schema`.

Making it a protocol makes it more expressive about precisely what it needs to do.
This can especially be helpful if/when we decide to replace Marshmallow with something else. 

Having said all. We don't use protocols in this code base, and I don't think these are super compelling reasons to introduce this new pattern when `Schema` works just as well. 😄

Reply:
Addressing the original question.
`.to_json_dict` is on the Config object
`.dump` is on the Schema object

`schema.load` should apply the schema validation.
https://marshmallow.readthedocs.io/en/stable/marshmallow.schema.html#marshmallow.schema.Schema.load